sql/core/src/main/scala/org/apache/spark/sql/UDFRegistration.scala

```diff
@@ -412,6 +419,63 @@ class UDFRegistration private[sql] (functionRegistry: FunctionRegistry) extends
   //////////////////////////////////////////////////////////////////////////////////////////////
 
   /**
+   * Register a Java UDF class
+   * @param name
+   * @param className
+   * @param returnType
+   */
+  def registerJava(name: String, className: String, returnType: DataType): Unit = {
```

Comment:
I don't know if we want to expose this API for general use - maybe make it private so that it can only be called from Python? And maybe update the scaladoc to something like "Register a Java UDF class using reflection - for use from Python".